skimage/feature/blob.py

```diff
@@ -405,16 +434,21 @@ def blob_doh(image, min_sigma=1, max_sigma=30, num_sigma=10, threshold=0.01,
         sigma_list = np.linspace(min_sigma, max_sigma, num_sigma)
 
     hessian_images = [_hessian_matrix_det(image, s) for s in sigma_list]
-    image_cube = np.dstack(hessian_images)
+
+    # Replace by image_cube = np.stack(hessian_images, axis=-1)
+    # When we upgrade minimal requirements to NumPy 1.10
+    sl = (slice(None),) * image.ndim + (np.newaxis,)
+    arrays = [np.asanyarray(arr) for arr in hessian_images]
+    extended_arrays = [arr[sl] for arr in arrays]
+    image_cube = np.concatenate(extended_arrays, axis=-1)
```

Comment:
I first thought that I could port it to 3D as well :-). But ``_hessian_matrix_det`` doesn't take 3D arguments. OK, I'll revert to the more simple syntax here, for simplicity.